baseapp/baseapp.go

```diff
@@ -949,11 +977,55 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 	// only update state if all messages pass
 	if result.IsOK() {
 		msCache.Write()
+		commitUncheckedFiles(ctx)
+	} else {
+		if mode == runTxModeDeliver {
+			deleteUncheckedFiles(ctx)
+		}
 	}
 
 	return
 }
 
+func copyFile(destination string, source string) error {
+	sourceFile, err := os.OpenFile(source, os.O_RDONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer sourceFile.Close()
+
+	destinationFile, err := os.OpenFile(destination, os.O_WRONLY | os.O_CREATE | os.O_APPEND, 0644)
+	if err != nil {
+		return err
+	}
+	defer destinationFile.Close()
+
+	_, err = io.Copy(destinationFile, sourceFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func commitUncheckedFiles(ctx sdk.Context) {
+	for _, key := range []string{"delegations", "unbond", "balance", "rewards"} {
+		err := copyFile(fmt.Sprintf("./extract/progress/%s.%d.%s", key, ctx.BlockHeight(), ctx.ChainID()), fmt.Sprintf("./extract/unchecked/%s.%d.%s", key, ctx.BlockHeight(), ctx.ChainID()))
+		if err != nil {
+			// TODO: Handle error instead of panicking
+			panic(err)
```

Comment:
we can't panic here on file not found; the source files often won't exist!

Reply:
fixed